### PR TITLE
Remove Stage1 compilation

### PR DESCRIPTION
**Pull request by piotrpMSFT**
@eerhardt @livarcocc @jgoshi @jonsequitur @krwq

### CODE REVIEW

File: build/Microsoft.DotNet.Cli.Compile.targets [hunk 1]
```diff
@@ -15,44 +15,16 @@
...
-        <StageSymbolsDirectory>$(Stage2SymbolsDirectory)</StageSymbolsDirectory>
-        <DotnetDir>$(Stage0Directory)</DotnetDir>
-      </Stage>
-    </ItemGroup>
-  </Target>
-
-  <Target Name="CompileStage"
-          DependsOnTargets="SetupStage;
-                            InitCrossgenProps;"
-          Inputs="%(Stage.Inputs)"
```
Comment (livarcocc) [inline build/Microsoft.DotNet.Cli.Compile.targets]: Are you sure it is safe to remove the incrementally check here? I wonder if this won't cause our signed dlls to be stomped over.

File: build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets [hunk 2]
```diff
@@ -107,7 +107,7 @@
     </Target>
 
     <Target Name="SetupNuGetPackagesArchiveInputsOutputs"
-            DependsOnTargets="Prepare;SetupStage;SetupAzureBlobInformation">
+            DependsOnTargets="Prepare;SetupAzureBlobInformation">
```
Comment (livarcocc) [inline build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets]: Make sure to run an archive generation to make verify this does not break anything. But looking at the code, this seems like left over.

File: build/Microsoft.DotNet.Cli.Compile.targets (hunk 1 above)
Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Compile.targets]: hahahah

File: build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets (hunk 2 above)
Comment (piotrpMSFT) [inline build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets]: I ran lzma archiver localy with no issues.

**Comment by piotrpMSFT** [comment]
@eerhardt I want to make sure this doesn't get in the way of your existing work.

**Comment by eerhardt** [comment]
> @eerhardt I want to make sure this doesn't get in the way of your existing work.

I don't have any real outstanding work in the CLI at the moment.  I was going to start working on the merge with the "new3" stuff, but taking this change is going to make it easier.

File: build/Microsoft.DotNet.Cli.Compile.targets (hunk 1 above)
Comment (eerhardt) [inline build/Microsoft.DotNet.Cli.Compile.targets]: In all honesty, do we still want incremental checks for `Compile`?

**Pull request merged**